### PR TITLE
Store registered URLs

### LINKED ISSUE #5: Store registered URLs
* Simple storage, e.g. SQLite?
* We could also retrieve them through a SPARQL query against the RDF store (`WHERE ?s a schema:Dataset`).

### PR DESCRIPTION
**Pull request by ddeboer**
Fix #5.

### CODE REVIEW

File: src/store.ts [hunk 1]
```diff
@@ -93,16 +103,42 @@ export class GraphDbDataStore implements Store {
...
+    const foundAtUrlQuads = [
+      factory.quad(
+        factory.namedNode(url.toString()),
+        factory.namedNode('http://schema.org/mainEntity'),
+        factory.namedNode(datasetUri),
+        factory.namedNode(this.registrationsGraph)
+      ),
+      factory.quad(
+        factory.namedNode(url.toString()),
+        factory.namedNode('http://schema.org/datePosted'),
```
Comment (ddeboer) [inline src/store.ts:133]: @coret I use `schema:mainEntity` and `schema:datePosted` (for the date at which the URL was last registered). Do these predicates make sense to you?

Comment (coret) [reply src/store.ts:133]: I have not found a vocabulary which has all the properties we need (as defined in 'Registered URL’s of dataset descriptions and catalogs' of the Design). As this data is for internal use only, I think we can define our own vocabulary like:
```
<https://www.openarch.nl/datasets/hga> a nde:CrawlResource
	nde:status		"valid"				(or invalid, deleted)
	nde:dateAdded 		....(DateTime)
	nde:dateLastFetch 	....(DateTime)
	nde:resourceType  	"dataset"  			(or "datasets", "datacatalog")
	nde:encodingFormat	"inlined-json-ld"		(or "microdata", "json-ld", "turtle", "nt", "rdf-xml")
	nde:foundVocabulary	"schema"			(or "dcat")
	nde:datasetsExtracted	<https://www.openarch.nl/id/dataset_hga>
```
When a CrawlResource, specified by @id (=url), already exists, the existing CrawlResource should not be updated nor should a new one be created.

Comment (ddeboer) [reply src/store.ts:133]: It’s good practice to prefer re-using existing ontologies over inventing your own.

In this case, I agree that Schema.org doesn’t fully match what we want. However, you have some redundant properties that can also change between crawls and therefore need to be checked each time anyway (so don’t need to be stored): 
`nde:resourceType` (we store datasets only), `nde:encodingFormat` and `nde:foundVocabulary` (only interesting for internal statistics). 

We’re then left with some properties for which Schema ‘equivalents’ do exist:

- `nde:dateLastFetch` = `schema:dateRead`
- `nde:dateAdded` = `schema:datePosted`
- `nde:datasetsExtracted` = `schema:mainEntity` or `schema:about`.

Comment (ddeboer) [reply src/store.ts:133]: `https://schema.org/status`

**Pull request merged**